internal/microsoft/data_microsoft_graph.go

```diff
@@ -0,0 +1,129 @@
+package microsoft
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/plugin/plugindata"
+)
+
+func makeMicrosoftGraphDataSource(loader MicrosoftGraphClientLoadFn) *plugin.DataSource {
+	return &plugin.DataSource{
+		Doc:      "The `microsoft_graph` data source queries Microsoft Graph.",
+		DataFunc: fetchMicrosoftGraph(loader),
+		Config: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Doc:         "The Azure client ID",
+					Name:        "client_id",
+					Type:        cty.String,
+					Constraints: constraint.RequiredNonNull,
+				},
+				{
+					Doc:    "The Azure client secret. Required if private_key_file/privat_key/cert_thumbprint is not provided.",
+					Name:   "client_secret",
+					Type:   cty.String,
+					Secret: true,
+				},
+				{
+					Doc:         "The Azure tenant ID",
+					Name:        "tenant_id",
+					Type:        cty.String,
+					Constraints: constraint.RequiredNonNull,
+				},
+				{
+					Doc:  "The path to the private key file. Ignored if private_key/client_secret is provided.",
+					Name: "private_key_file",
+					Type: cty.String,
+				},
+				{
+					Doc:  "The private key contents. Ignored if client_secret is provided.",
+					Name: "private_key",
+					Type: cty.String,
+				},
+				{
+					Doc:  "The certificate thumbprint. Ignored if client_secret is provided.",
```

Comment:
@traut go client library doesn't support certificate thumbprint. Please check:
https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/4961ef926304f2aa240b1751c4b87f83145356d2/apps/design/design.md#use-of-x509certificate-and-certfrompem-function

We can remove it and let users provide a private key which includes the certificate. wdyt?

Reply:
Interesting! There is a certificate thumbprint available in Azure UI --
<img width="312" alt="Screenshot 2024-09-22 at 20 13 06" src="https://github.com/user-attachments/assets/66d69c5c-f9ae-49da-b142-14fcf5fc586b">

Of course, if it's unnecessary, there is no reason to ask for it!